src/Tizen.NUI.Wearable/src/public/CircularSlider.cs

```diff
@@ -0,0 +1,674 @@
+        public float MinValue
+        {
+            get
+            {
+                return (float)GetValue(MinValueProperty);
+            }
+            set
+            {
+                SetValue(MinValueProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// The property to get/set the current value of the CircularSlider.
+        /// The default value is 0.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public float CurrentValue
+        {
+            get
+            {
+                return (float)GetValue(CurrentValueProperty);
+            }
+            set
+            {
+                // For the first Animation effect
+                sweepAngleAnimation = AnimateVisual(progressVisual, "sweepAngle", progressVisual.SweepAngle, 0, 100, AlphaFunction.BuiltinFunctions.EaseIn);
+
+                SetValue(CurrentValueProperty, value);
+
+                UpdateAnimation();
+            }
+        }
+
+        /// <summary>
+        /// The property to get/set Track object color of the CircularSlider.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Color TrackColor
+        {
+            get
+            {
+                return (Color)GetValue(TrackColorProperty);
+            }
+            set
+            {
+                SetValue(TrackColorProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// The property to get/set Progress object color of the CircularSlider.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Color ProgressColor
+        {
+            get
+            {
+                return (Color)GetValue(ProgressColorProperty);
+            }
+            set
+            {
+                SetValue(ProgressColorProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the size of the thumb of Slider.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Size ThumbSize
+        {
+            get
+            {
+                return (Size)GetValue(ThumbSizeProperty);
+            }
+            set
+            {
+                SetValue(ThumbSizeProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// The property to get/set Thumb object color of the CircularSlider.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Color ThumbColor
+        {
+            get
+            {
+                return (Color)GetValue(ThumbColorProperty);
+            }
+            set
+            {
+                SetValue(ThumbColorProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// Flag to be enabled or disabled in CircularSlider.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool IsEnabled
```

Comment:
What do you think of override `View.OnControlStateChanged` instead?

`CostrolState` will be set by API User. Then, `CircularSlider` can handle that state.
```
circularSlider.ControlState = ControlStates.Disabled;
or
circularSlider.ControlState = ControlStates.Normal;
```